internal/manager/telemetry/manager_test.go

```diff
@@ -0,0 +1,407 @@
+		telemetry.OptManagerPeriod(time.Hour),
+		telemetry.OptManagerLogger(logr.Discard()),
+	)
+	require.NoError(t, err)
+	require.NotNil(t, mgr)
+
+	ch := make(chan []byte)
+	consumer := telemetry.NewConsumer(
+		serializers.NewSemicolonDelimited(),
+		forwarders.NewChannelForwarder(ch),
+	)
+	require.NoError(t, mgr.AddConsumer(consumer))
+
+	require.NoError(t, mgr.Start())
+	defer mgr.Stop()
+	require.NoError(t, mgr.TriggerExecute(ctx, "test-signal"))
+	select {
+	case b := <-ch:
+		require.Equal(t,
+			fmt.Sprintf(
+				"<14>"+
+					"signal=test-signal;"+
+					"db=off;"+
+					"feature-gateway=true;"+
+					"feature-knative=false;"+
+					"hn=%s;"+
+					"kv=3.1.1;"+
+					"uptime=0;"+
+					"k8s_arch=linux/amd64;"+
+					"k8s_provider=UNKNOWN;"+
+					"k8sv=v1.24.5;"+
+					"k8sv_semver=v1.24.5;"+
+					"k8s_nodes_count=4;"+
+					"k8s_pods_count=8;"+
+					"k8s_services_count=17;"+
+					"kinm=c3,l2,l3,l4;"+
+					"mdep=i3,k3,km3,l3,t3;"+
+					"mdist=all17,c1,i2,k1,km1,l2,t1;"+
+					"\n",
+				hostname),
+			string(b),
+		)
+	case <-time.After(time.Second):
+		t.Fatal("we should get a report but we didn't")
+	}
+}
+
+func prepareScheme(t *testing.T) *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	require.NoError(t, testk8sclient.AddToScheme(scheme))
+	// This doesn't work :(
+	// https://kubernetes.slack.com/archives/C0EG7JC6T/p1676125039126589
+	require.NoError(t, gatewayv1beta1.Install(scheme))
+	return scheme
+}
+
+func prepareObjects(pod apitypes.NamespacedName) []runtime.Object {
+	setRandomUUIDName := func(o client.Object) runtime.Object {
+		o.SetName(uuid.NewString())
+		return o
+	}
+
+	return []runtime.Object{
+		&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: pod.Namespace,
+				Name:      pod.Name,
+				Annotations: map[string]string{
+					"linkerd.io/proxy-version": "1.0.0",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "ingress-controller",
+						Image: "kong/kubernetes-ingress-controller:2.8",
+					},
+					{
+						Name:  "kong",
+						Image: "kong/kong:3.1.1",
+					},
+					// sidecars
+					{Name: "linkerd-proxy"},
+					{Name: "envoy-sidecar"},
+				},
+				// init containers.
+				InitContainers: []corev1.Container{
+					{Name: "linkerd-init"},
+				},
+			},
+		},
+		// service with no endpoints.
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: pod.Namespace,
+				Name:      "kong-proxy",
+			},
+		},
+		// endpoints.
+		&corev1.Endpoints{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: pod.Namespace,
+				Name:      "kong-proxy",
+			},
+			Subsets: []corev1.EndpointSubset{
+				{
+					Addresses: []corev1.EndpointAddress{
+						{
+							TargetRef: &corev1.ObjectReference{
+								Kind:      "Pod",
+								Namespace: pod.Namespace,
+								Name:      pod.Name,
+							},
+						},
+					},
+				},
+			},
+		},
+		// &gatewayv1beta1.Gateway{},
```

Comment:
Could we uncomment it and assert that the count of gateways gets reported as well?

Reply:
This unfortunately doesn't work because ... pluralizing in English is hard stuff: https://github.com/kubernetes/kubernetes/pull/110053.

Related thread in #sig-api-machinery that I started where there's no solution yet 😬 
https://kubernetes.slack.com/archives/C0EG7JC6T/p1676125039126589

I'll add some more comments later on so that it's clear why can't keep this.